matrixConsole/AppDelegate.m

```diff
@@ -423,6 +425,52 @@ - (void)application:(UIApplication*)application didReceiveRemoteNotification:(NS
 
 - (void)initMatrixSessions
 {
+    [MXKAccount registerOnCertificateChangeBlock:^BOOL(MXKAccount *mxAccount, NSData *certificate) {
+        
+        __block BOOL isTrusted;
+        dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+        
+        MXCredentials *mxCredentials = mxAccount.mxCredentials;
+        
+        NSString *title = [NSBundle mxk_localizedStringForKey:@"ssl_could_not_verify"];
+        NSString *existing_expl = mxCredentials.allowedCertificate ? [NSBundle mxk_localizedStringForKey:@"ssl_expected_existing_expl"] : [NSBundle mxk_localizedStringForKey:@"ssl_unexpected_existing_expl"];
+        NSString *homeserverURLStr = [NSString stringWithFormat:[NSBundle mxk_localizedStringForKey:@"ssl_homeserver_url"], mxCredentials.homeServer];
+        NSString *fingerprint = [NSString stringWithFormat:[NSBundle mxk_localizedStringForKey:@"ssl_fingerprint_hash"], @"SHA256"];
+        NSString *certFingerprint = [certificate SHA256AsHexString];
```

Comment:
Oh, I see - this is when the cert changes. Why is one handled by the app and one in MatrixKit?